Organize labels

See discussion starting at https://github.com/webcompat/webcompat.com/issues/402#issuecomment-63261824

Also hide status/step/progress category from labels.

First let's fix the test.

```
→ nosetests
............F...
======================================================================
FAIL: Test that the new filtering is correct.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/karl/code/webcompat.com/tests/test_urls.py", line 106, in test_issues_new
    self.assertEqual(filter_new(issues), result)
AssertionError: '[{"labels": [{"name": "bug"}, {"name": "help wanted"}], "id": 0, "title": "fake bug 0"}, {"labels": [], "id": 1, "title": "fake bug 1"}, {"labels": [{"name": "contactready"}], "id": 2, "title": "fake bug 2"}, {"labels": [{"name": "needsdiagnosis"}], "id": 3, "title": "fake bug 3"}, {"labels": [{"name": "needscontact"}], "id": 4, "title": "fake bug 4"}, {"labels": [{"name": "sitewait"}], "id": 5, "title": "fake bug 5"}]' != '[{"labels": [{"name": "bug"}, {"name": "help wanted"}], "id": 0, "title": "fake bug 0"}, {"labels": [], "id": 1, "title": "fake bug 1"}]'

----------------------------------------------------------------------
Ran 16 tests in 0.703s

FAILED (failures=1)
```

--- a/webcompat/static/js/lib/issue-list.js
+++ b/webcompat/static/js/lib/issue-list.js
@@ -293,7 +293,7 @@ issueList.IssueView = Backbone.View.extend({
   events: {
     'click .js-issue-label': 'labelSearch',
   },
-  _filterRegex: /&*stage=(new|needscontact|needsdiagnosis|contactready|sitewait|closed)&*/i,
+  _filterRegex: /(stage=(?:new|status-needscontact|status-needsdiagnosis|status-contactready|status-sitewait|status-closed))/ig,
   _isLoggedIn: $('body').data('username'),
   _loadingIndicator: $('.js-loader'),
   _nextButton: $('.js-pagination-next'),

--- a/webcompat/static/js/lib/labels.js
+++ b/webcompat/static/js/lib/labels.js
@@ -31,7 +31,7 @@ issues.LabelsView = Backbone.View.extend({
   subTemplate: _.template([
     '<% _.each(labels, function(label) { %>',
       '<span class="Label Label--badge" style="background-color:#<%=label.color%>">',
-        '<%= label.name %>',
+        '<%= label.name.replace(/(browser|status)-/, "") %>',
       '</span>',
     '<% }); %>'].join('')),
   render: function() {

--- a/webcompat/static/js/lib/models/issue.js
+++ b/webcompat/static/js/lib/models/issue.js
@@ -97,7 +97,6 @@
         labelsArray[i] = 'os-'.concat(labelsArray[i])
       };
     };
-    console.log(labelsArray);
     var self = this;
     if (!$.isArray(labelsArray)) {
       return;

--- a/webcompat/static/js/lib/models/issue.js
+++ b/webcompat/static/js/lib/models/issue.js
@@ -83,6 +83,21 @@
     });
   },
   updateLabels: function(labelsArray) {
+    // maybe this should be in a shared config file outside of python/JS
+    var statusLabels = ['contactready', 'needscontact', 'needsdiagnosis', 'sitewait', ' closed-duplicate', 'closed-fixed', 'closed-invalid'];
+    var browserLabels = ['chrome', 'firefox', 'ie', 'opera', 'safari', 'vivaldi'];
+    var osLabels = ['android', 'fxos', 'ios', 'linux', 'mac', 'win']
+    // we check if we need to append the correct string before sending stuff back
+    for (var i = labelsArray.length - 1; i >= 0; i--) {
+      if (statusLabels.indexOf(labelsArray[i]) != -1) {
+        labelsArray[i] = 'status-'.concat(labelsArray[i])
+      } else if (browserLabels.indexOf(labelsArray[i]) != -1) {
+        labelsArray[i] = 'browser-'.concat(labelsArray[i])
+      } else if (osLabels.indexOf(labelsArray[i]) != -1) {
+        labelsArray[i] = 'os-'.concat(labelsArray[i])
+      };
+    };
+    console.log(labelsArray);
     var self = this;
     if (!$.isArray(labelsArray)) {
       return;

--- a/webcompat/templates/issue-list.html
+++ b/webcompat/templates/issue-list.html
@@ -75,7 +75,7 @@
             </div>
             <div class="wc-IssueItem-section">
               <span class="wc-IssueItem-label js-issue-label"><% _.each(issue.labels, function(label) { %>
-                <a href="#" class="wc-Labels" title="Labels : <%= label.name %>"><%= label.name %></a>
+                <a href="#" class="wc-Labels" title="Labels : <%= label.name.replace(/(browser|status)-/, '') %>"><%= label.name.replace(/(browser|status)-/, '') %></a>
               <% }); %></span>
             </div>
           </div>

--- a/webcompat/templates/issue.html
+++ b/webcompat/templates/issue.html
@@ -30,7 +30,7 @@
           <span class="Label-list">
           <% _.each(labels, function(label) { %>
             <span class="Label Label--badge" style="background-color:#<%=label.color%>">
-              <%= label.name %>
+              <%= label.name.replace(/(browser|status)-/, '') %>
             </span>
           <% }); %>
           </span>
@@ -118,9 +118,9 @@
           <% _.each(labels, function(label) { %>
             <label class="LabelEditor-item">
               <span class="LabelEditor-color" style="background-color:#<%=label.color%>">
-                <input class="LabelEditor-checkbox" type="checkbox" name="<%= label.name %>" data-color="<%=label.color%>">
+                <input class="LabelEditor-checkbox" type="checkbox" name="<%= label.name.replace(/(browser|status)-/, '') %>" data-color="<%=label.color%>">
               </span>
-              <span class="LabelEditor-name"><%= label.name %></span>
+              <span class="LabelEditor-name"><%= label.name.replace(/(browser|status)-/, '') %></span>
               </label>
             <% }); %>
           </div>

